aya/src/programs/links.rs

```diff
@@ -106,17 +79,17 @@ impl<T: Link> Drop for LinkMap<T> {
 
 /// The identifier of an `FdLink`.
 #[derive(Debug, Hash, Eq, PartialEq)]
-pub struct FdLinkId(pub(crate) RawFd);
+pub struct FdLinkId(pub(crate) Option<RawFd>);
 
 /// A file descriptor link.
 #[derive(Debug)]
 pub struct FdLink {
-    pub(crate) fd: RawFd,
+    pub(crate) fd: Option<RawFd>,
 }
 
 impl FdLink {
     pub(crate) fn new(fd: RawFd) -> FdLink {
-        FdLink { fd }
+        FdLink { fd: Some(fd) }
     }
 
     /// Pins the FdLink to a BPF filesystem.
```

Comment:
```
Pins the link to a BPF file system.

When a link is pinned to a BPF file system, it will remain attached even after the link instance is dropped, and will only be detached once the pinned file is removed. To unpin, see [PinnedFd::unpin].

The parent directories in the provided path must already exist before calling this method, and the path must point to a location on a BPF file system.
```